ResidentInformationApi/V1/UseCase/ListContactsUseCase.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using ResidentInformationApi.V1.Boundary.Requests;
+using ResidentInformationApi.V1.Boundary.Responses;
+using ResidentInformationApi.V1.Factories;
+using ResidentInformationApi.V1.Gateways;
+
+namespace ResidentInformationApi.V1.UseCase
+{
+    public class ListContactsUseCase : IListContactsUseCase
+    {
+        private readonly IAcademyInformationGateway _academyGateway;
+        private readonly IHousingInformationGateway _housingGateway;
+        private readonly IMosaicInformationGateway _mosaicGateway;
+
+        public ListContactsUseCase(
+            IAcademyInformationGateway academyGateway,
+            IHousingInformationGateway housingGateway,
+            IMosaicInformationGateway mosaicGateway
+            )
+        {
+            _academyGateway = academyGateway;
+            _housingGateway = housingGateway;
+            _mosaicGateway = mosaicGateway;
+        }
+
+        public async Task<ResidentInformationResponse> Execute(ResidentQueryParam rqp)
+        {
+            var academyClaimants = await _academyGateway.GetClaimantInformation(rqp).ConfigureAwait(true);
+            var housingResidents = await _housingGateway.GetResidentInformation(rqp).ConfigureAwait(true);
+            var mosaicResidents = await _mosaicGateway.GetResidentInformation(rqp).ConfigureAwait(true);
+
+            var academyResults = academyClaimants.Select(x =>
+                new ResidentInformationResult
+                {
+                    System = "Academy",
+                    SystemId = x.ClaimId.ToString(),
+                    SystemUrl = new Uri("https://academy-api.hackney.gov.uk"),
+                    Data = x.ToResponse()
+                });
+
+            var housingResults = housingResidents.Select(x =>
+                new ResidentInformationResult
+                {
+                    System = "Housing",
+                    SystemId = x.HouseReference.ToString(),
+                    SystemUrl = new Uri("https://housing-api.hackney.gov.uk"),
```

Comment:
Don't think we should have DNS hard-coded.
It's likely that different environments would have a different dns (i.e. academy-api-test.hackney.gov.uk).
Or indeed ALL apis might have the same DNS per environment i.e. https://cloudapis.hackney.gov.uk)
This should probably be config - env variable.

Reply:
Thanks for the pointers, I've updated this!